pcdsutils/import_timer.py

```diff
@@ -0,0 +1,370 @@
+"""
+Helpers for timing how long it takes to do various imports.
+"""
+from __future__ import annotations
+
+import argparse
+import subprocess
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+from prettytable import PrettyTable
+
+
+@dataclass(frozen=True)
+class ImportTimeStats:
+    """
+    Stats about importing this module or submodule.
+
+    Attributes
+    ----------
+    module : str
+        The full import name, e.g. pcdsdevices.signal
+    root_module : str
+        The source module, e.g. pcdsdevices
+    self_time_raw : int
+        The time it took to import this module in microseconds, not counting
+        the time it took to import other modules.
+    self_time : float
+        The time it took to import this module in seconds, not counting
+        the time it took to import other modules.
+    cumulative_time_raw : int
+        The time it took to import this module in microseconds,
+        including the time it took to import other modules.
+    cumulative_time : float
+        The time it took to import this module in seconds,
+        including the time it took to import other modules.
+    indent_level : int
+        How deep this import was. An indent level of 0 indicates that
+        this was the main import or a python internal import used to set up
+        the import system itself. Each additional level means that this module
+        was imported by a module one level lower. For example, a module with
+        indent level 2 was imported by a module with import level 1.
+    """
+    module: str
+    root_module: str
+    self_time_raw: int
+    self_time: float
+    cumulative_time_raw: int
+    cumulative_time: float
+    indent_level: int
+
+    @classmethod
+    def from_line(cls, line: str) -> ImportTimeStats:
+        """
+        Assemble an ImportTimeStats from a python -X importtime output line.
+
+        Example lines:
+        import time:      2848 |       2848 |     pcdsdevices.stopper
+        import time:      7611 |      10459 |   pcdsdevices.valve
+        import time:      1859 |       1859 |   pcdsdevices.wfs
+        import time:      2913 |   12317383 | pcdsdevices.device_types
+        """
+        if not line.startswith('import time'):
+            raise ValueError(
+                f'Recieved {line}, which is not a valid importtime output.'
+            )
+        line = line.split('import time:')[1]
+        self_time_raw, cumulative_time_raw, module = line.split('|')
+        self_time_raw = int(self_time_raw.strip())
+        cumulative_time_raw = int(cumulative_time_raw.strip())
+        indent_level = (module.count(' ') - 1) / 2
```

Comment:
This is float division (does it make sense here?)
```suggestion
        indent_level = (module.count(' ') - 1) // 2
```

Reply:
Right, I know that for normal output that this will always come out to an integer, but it's good to be thorough

Reply:
The table gave me pause:
<img width="147" alt="image" src="https://user-images.githubusercontent.com/5139267/174906994-63f3e6e0-666c-46dd-b6ab-649e077f3006.png">

Is this just an artifact of PrettyTable rendering - or do we actually have floats?

Reply:
it must become a float, I wasn't being careful enough